src/Hypergrid.js

```diff
@@ -1550,6 +1550,20 @@ Hypergrid.prototype = {
         });
         this.canvas.dispatchEvent(clickEvent);
     },
+    /**
+     * @memberOf Hypergrid.prototype
+     * @param {number} c - grid column index.
+     * @param {string[]} keys
+     */
+    fireSyntheticColumnSortEvent: function(c, keys) {
```

Comment:
We will need the grid to trigger the events. The application developer will listen to these events and decide to act (or not) and if acting means doing a setData or applyAnaltyics